### ISSUE TITLE
Support TensorRT-LLM?

### ISSUE THREAD

**Issue by npuichigo** [langchain-ai/langchain#12474]
### Feature request

Any plan to support nvidia's latest TensorRT-LLM, maybe via triton-inference-server backend?

### Motivation

New integration

### Your contribution

Test

**Comment by npuichigo** [comment]
I invented a wheel to provide TensorRT LLM an OpenAI compatible API. Welcome to have a try https://github.com/npuichigo/openai_trtllm

**Comment by fciannella** [comment]
We have an official NVIDIA connector now. Still WIP: https://pypi.org/project/langchain-nvidia-trt/

**Comment by mkhludnev** [comment]
> We have an official NVIDIA connector now. Still WIP: https://pypi.org/project/langchain-nvidia-trt/

Thanks, @fciannella. Is https://github.com/langchain-ai/langchain/tree/master/libs/partners/nvidia-trt/langchain_nvidia_trt the source code for this package?

**Comment by fciannella** [comment]
Yes!

We are still working on a bug fix before we push more documentation and
announce it officially.

Feel free to provide any feedback!

**Comment by mkhludnev** [comment]
> Feel free to provide any feedback!

Ok. Thanks. I evaluated it in my env. Here we go:

- verbose flag would be quite helpful to propagate for debugging  
- there's `cuda-python` dependency but there's no need in it for client access, and no way to install it on macos. I'm not sure about the tests, but code itself just runs. 
- when it receives an exception it fails on casting to string 
- the hardest problem - adjusting to the model parameters, I had to edit llms.py, but presumably (don't know exactly) the code might obtain the model's parameters via API.

**Comment by mkhludnev** [comment]
@fciannella btw https://github.com/langchain-ai/langchain/pull/16749 is worth a release!

**Comment by mkhludnev** [comment]
I use this code with vLLM model deployed with 

```
backend: "vllm"

# Disabling batching in Triton, let vLLM handle the batching on its own.
max_batch_size: 0

# We need to use decoupled transaction policy for saturating
# vLLM engine for max throughtput.
model_transaction_policy {
  decoupled: True
}
```

It seems ok, but it doesn't support explicit stop signal. https://github.com/langchain-ai/langchain/blob/849051102a6e315072e3a1d8dfdcee1527436c92/libs/partners/nvidia-trt/langchain_nvidia_trt/llms.py#L299
I had to change https://github.com/langchain-ai/langchain/blob/849051102a6e315072e3a1d8dfdcee1527436c92/libs/partners/nvidia-trt/langchain_nvidia_trt/llms.py#L218 to `force_batch=True,` there's to argument to turn it properly.

**Comment by mkhludnev** [comment]
The following problem resides not it this code base, but in integration technology. 

## TL;DR
Triton REST seems fine, but gRPRC, which is used by this code, seems prone to trimming LLM response. 

It's a problem when triton runs vLLM, which [concatenates prompt and LLM response](https://github.com/triton-inference-server/vllm_backend/blob/52c1c3ce7500581469150e38f61b3a200ac3e82a/src/model.py#L216) (btw why?), where prompt might be really long in RAG.
Heres the triton REST response
```
{"model_name":"vllm_model","model_version":"1","text_output":"<|system|>\nUse the following pieces of ... really long prompt..\n</s>\n<|assistant|>\nNo, you should not feed your dog snail as it is not a safe food for dogs. Snails can carry parasites ... full long and informative response ...  for humans, the response remains valid."}    
```
And when I send the same request via gRPC with this library, I have the same long prompt repeated in response and then quite a short excerpt of LLM response remains. Here's verbose log trace, not a function function return:
```
 raw_output_contents: "\224\004\000\000\n<|system|>\nUse the following pieces of i ....l?\n</s>\n<|assistant|>\nNo, it is not safe to feed a dog with snails. While some"   <-   the most valuable part has been trimmed! 
```
 PS I beg pardon for such a silly question 🐕 🐌

**Comment by jasonngap1** [comment]
> We have an official NVIDIA connector now. Still WIP: https://pypi.org/project/langchain-nvidia-trt/

Hi, are there any updates on when this integration will be released please?
